### PR TITLE
Completed sgt-back-end challenge - Francis Manalang

### PR DESCRIPTION
**Pull request by francismanalang**
[Screen Recording 2022-08-26 at 10.38.59 AM.zip](https://github.com/francismanalang/c0622-code-solutions/files/9435431/Screen.Recording.2022-08-26.at.10.38.59.AM.zip)
[Screen Recording 2022-08-26 at 10.59.11 AM.zip](https://github.com/francismanalang/c0622-code-solutions/files/9435438/Screen.Recording.2022-08-26.at.10.59.11.AM.zip)
Uploading Screen Recording 2022-08-26 at 11.01.07 AM.mov…

### CODE REVIEW

**Comment by francismanalang** [comment]
https://user-images.githubusercontent.com/101234537/186965635-ac684f54-6498-4f57-b4f5-45218fc1f4c5.mov

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  if (!name || !course || !score) {
+    res.status(400).json({ error: 'Invalid or missing. name, course and score are required' });
+  } else if (!Number.isInteger(score) || score <= 0 || score > 100) {
+    res.status(400).json({ error: 'Please provide a valid score between 1-100' });
+    return;
+  }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: A `score` of `0` should not result in a `400 Bad Request`.

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  `;
+
+  db.query(sql, params)
+    .then(result => {
+      const deletedGrade = result.rows[0];
+      if (!deletedGrade) {
+        res.status(404).json({ error: `Cannot find grade with gradeId ${gradeId}` });
+      } else {
+        res.status(204).json(deletedGrade);
+      }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: A `204 No Content` response should not contain content.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const name = req.body.name;
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  if (!name || !course || !score) {
+    res.status(400).json({ error: 'Invalid or missing. name, course and score are required' });
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: > A `score` of `0` should not result in a `400 Bad Request`.

Please address this issue.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  const course = req.body.course;
+  const score = Number(req.body.score);
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({ error: '"gradeId" must be a positive integer' });
+  } else if (!name || !course || score === undefined) {
+    res.status(400).json({ error: 'Invalid or missing. name, course and score are required.' });
+  } else if (score < 0 || score > 100) {
+    res.status(400).json({ error: 'Please provide a score between 0-100' });
+  }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js]: Scores also need to be an integer

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: sgt-back-end/index.js (hunk 4 above)
Comment (TimelikeClosure) [reply sgt-back-end/index.js]: @francismanalang It looks like you addressed the POST, now please address the PUT

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**